src/coreclr/nativeaot/Runtime/CMakeLists.txt

```diff
@@ -153,11 +153,17 @@ else()
     ${GC_DIR}/unix/cgroup.cpp
   )
 
-  if(CLR_CMAKE_TARGET_ARCH_WASM)
+  if (CLR_CMAKE_TARGET_ARCH_WASM)
     list(APPEND COMMON_RUNTIME_SOURCES
       ${GC_DIR}/wasm/gcenv.wasm.cpp
       unix/cgroupcpu.cpp
     )
+    if (CLR_CMAKE_TARGET_WASI)
+      list(APPEND COMMON_RUNTIME_SOURCES
+        wasm/wasi.cpp
+        wasm/wasi.c
```

Comment:
Are two files important (can we have one/one for each logical polyfill area)?

Reply:
Have separated into logical areas